webhooks/virtualmachinesnapshot/validation/virtualmachinesnapshot_validator.go

```diff
@@ -56,10 +62,47 @@ func (v validator) For() schema.GroupVersionKind {
 	return vmopv1.GroupVersion.WithKind(reflect.TypeOf(vmopv1.VirtualMachineSnapshot{}).Name())
 }
 
+// ValidateCreate makes sure the VirtualMachineSnapshot create request is valid.
 func (v validator) ValidateCreate(ctx *pkgctx.WebhookRequestContext) admission.Response {
-	// TODO: Implement validation for create
+	vmSnapshot, err := v.vmSnapshotFromUnstructured(ctx.Obj)
+	if err != nil {
+		return webhook.Errored(http.StatusBadRequest, err)
+	}
+
+	var fieldErrs field.ErrorList
+
+	vmRefField := field.NewPath("spec", "vmRef")
+
+	if vmSnapshot.Spec.VMRef == nil {
+		fieldErrs = append(fieldErrs, field.Required(vmRefField, "vmRef must be provided"))
+	}
```

Comment:
You could just `} else {` this here to not check the same thing for not nil below